lib/chef/resource/registry_key.rb

```diff
@@ -152,47 +150,36 @@ class RegistryKey < Chef::Resource
       # may want to extend the state_attrs API with the ability to rename POST'd attrs.
       #
       # See lib/chef/resource_reporter.rb for more information.
+      @unscrubbed_values = []
       attr_reader :unscrubbed_values
 
-      def initialize(name, run_context = nil)
-        super
-        @values, @unscrubbed_values = [], []
-      end
+      VALID_VALUE_HASH_KEYS = %i{name type data}.freeze
 
       property :key, String, name_property: true
 
-      VALID_VALUE_HASH_KEYS = %i{name type data}.freeze
-
-      def values(arg = nil)
-        if not arg.nil?
-          if arg.is_a?(Hash)
-            @values = [ Mash.new(arg).symbolize_keys ]
-          elsif arg.is_a?(Array)
-            @values = []
-            arg.each do |value|
-              @values << Mash.new(value).symbolize_keys
-            end
+      property :values, [Hash, Array, nil],
+        default: [],
+        coerce: proc { |v|
+          case v
+          when Hash
+            @unscrubbed_values = [ Mash.new(v).symbolize_keys ]
+          when Array
+            @unscrubbed_values = v.map { |value| Mash.new(value).symbolize_keys }
           else
-            raise ArgumentError, "Bad type for RegistryKey resource, use Hash or Array"
+            raise ArgumentError, "Bad type for RegistryKey resource, use Hash or Array" unless v.nil?
```

Comment:
this can be removed since the property won't accept anything other than a Hash or Array